src/browser/declarativewebutils.cpp

```diff
@@ -75,13 +77,14 @@ void DeclarativeWebUtils::clearStartupCacheIfNeeded()
 {
     QFileInfo systemStamp(gSystemComponentsTimeStamp);
     if (systemStamp.exists()) {
-        QString mostProfilePath = QDir::homePath() + gProfilePath;
-        QString localStampString(mostProfilePath + QString("/_CACHE_CLEAN_"));
-        QFileInfo localStamp(localStampString);
-        if (localStamp.exists() && systemStamp.lastModified() > localStamp.lastModified()) {
-            QDir cacheDir(mostProfilePath + "/startupCache");
-            cacheDir.removeRecursively();
-            QFile(localStampString).remove();
+        QString startupCacheDir = QDir::homePath() + gStartupCacheDir;
+        QDir cacheDir(startupCacheDir);
+        if (cacheDir.exists()) {
+            QFileInfoList files = cacheDir.entryInfoList(QDir::Files, QDir::Time);
+            if (files.count() > 0 && systemStamp.lastModified() > files.first().lastModified()) {
+                qDebug() << "Removing cache to synchronise with fresh embedlite-components-qt5";
```

Comment:
You can just use printf-like `qDebug("Removing cache to synchronise with fresh embedlite-components-qt5");` as it's quite simple here. Then you don't need to include `<QtCore/QDebug>` as well.

Reply:
Interesting, I wasn't aware this was possible. I've changed it.